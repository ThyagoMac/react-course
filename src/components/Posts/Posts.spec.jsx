import { render, screen } from "@testing-library/react";
import { Posts } from ".";

const props = {
  posts: [
    {
      id: 1,
      title: "Post Title Example1",
      body: "Post Body Example1",
      cover: "img/img1.png"
    },
    {
      id: 2,
      title: "Post Title Example2",
      body: "Post Body Example2",
      cover: "img/img2.png"
    },
    {
      id: 3,
      title: "Post Title Example3",
      body: "Post Body Example3",
      cover: "img/img3.png"
    },
  ]
}

describe("<Post />", () => {
  it("Render posts", () => {
    render(<Posts {...props} />);

    const postsHeadings = screen.getAllByRole("heading", { name: /title/i });
    const postsImgs = screen.getAllByRole("img", { name: /title/i });
    const postImgTwo = screen.getByRole("img", { name: /Title Example2/i });
    const p = screen.getAllByText(/body/i);

    expect(postImgTwo).toHaveAttribute("src", "img/img2.png");
    expect(postsImgs).toHaveLength(3);
    expect(postsHeadings).toHaveLength(3);
    expect(p).toHaveLength(3);

  });

  it("Should not render with empty posts", () => {
    render(<Posts />);;

    expect(screen.queryByRole("heading", { name: /title/i }))
      .not.toBeInTheDocument();
  });

  it("Should match snapshot", () => {
    const { container } = render(<Posts {...props} />);;

    expect(container.firstChild).toMatchSnapshot();
  });
});