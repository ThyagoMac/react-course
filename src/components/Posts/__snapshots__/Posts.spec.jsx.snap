// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Posts /> Should match snapshot 1`] = `
<div
  class="posts"
>
  <div
    class="post"
  >
    <img
      alt="Post Title Example1"
      src="img/img1.png"
    />
    <div
      class="post-content"
    >
      <h2>
        (
        1
        ) 
        Post Title Example1
      </h2>
      <p>
        Post Body Example1
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Post Title Example2"
      src="img/img2.png"
    />
    <div
      class="post-content"
    >
      <h2>
        (
        2
        ) 
        Post Title Example2
      </h2>
      <p>
        Post Body Example2
      </p>
    </div>
  </div>
  <div
    class="post"
  >
    <img
      alt="Post Title Example3"
      src="img/img3.png"
    />
    <div
      class="post-content"
    >
      <h2>
        (
        3
        ) 
        Post Title Example3
      </h2>
      <p>
        Post Body Example3
      </p>
    </div>
  </div>
</div>
`;
